cmd/all-in-one/main.go

```diff
@@ -287,32 +286,16 @@ func startQuery(
 	if err := server.Start(); err != nil {
 		svc.Logger.Fatal("Could not start jaeger-query service", zap.Error(err))
 	}
+
 	return server
 }
 
-func initTracer(svc *flags.Service) io.Closer {
-	logger := svc.Logger
-	traceCfg := &jaegerClientConfig.Configuration{
-		ServiceName: "jaeger-query",
-		Sampler: &jaegerClientConfig.SamplerConfig{
-			Type:  "const",
-			Param: 1.0,
-		},
-		RPCMetrics: true,
-	}
-	traceCfg, err := traceCfg.FromEnv()
-	if err != nil {
-		logger.Fatal("Failed to read tracer configuration", zap.Error(err))
-	}
-	tracer, closer, err := traceCfg.NewTracer(
-		jaegerClientConfig.Metrics(jlibadapter.NewAdapter(svc.MetricsFactory)),
-		jaegerClientConfig.Logger(jaegerClientZapLog.NewLogger(logger)),
-	)
+func initTracer() error {
+	tracer, err := jtracer.New()
 	if err != nil {
-		logger.Fatal("Failed to initialize tracer", zap.Error(err))
+		log.Fatal("Failed to initialize tracer", zap.Error(err))
 	}
-	opentracing.SetGlobalTracer(tracer)
-	return closer
+	return tracer.Close(context.Background())
```

Comment:
what do you expect to happen after this line?

Reply:
I think tracer should shutdown after the init. Or I should be returning tracer obj and get it closed inside `svc.RunAndThen(func() {`

Reply:
how do you expect to use the tracer later if you shut it down on init?